pkg/operator/ceph/cluster/cluster.go

```diff
@@ -143,20 +153,28 @@ func (c *cluster) createInstance(rookImage string) error {
 	}
 
 	// Start the mon pods
-	c.mons = mon.New(c.context, c.Namespace, c.Spec.DataDirHostPath, rookImage, c.Spec.CephVersion, c.Spec.Mon, cephv1.GetMonPlacement(c.Spec.Placement),
-		c.Spec.Network.HostNetwork, cephv1.GetMonResources(c.Spec.Resources), c.ownerRef)
-	clusterInfo, err := c.mons.Start()
+	c.mons = mon.New(c.Identity, c.context, c.Namespace,
+		c.Spec.DataDirHostPath, rookImage, c.Spec.CephVersion, c.Spec.Mon,
+		cephv1.GetMonPlacement(c.Spec.Placement), c.Spec.Network.HostNetwork,
+		cephv1.GetMonResources(c.Spec.Resources), c.ownerRef)
+	err = c.mons.Start()
 	if err != nil {
 		return fmt.Errorf("failed to start the mons. %+v", err)
 	}
 
+	// The cluster Identity must be established at this point
+	if !c.Identity.IsEstablished() {
```

Comment:
I want to be very explicit about the fact that the FSID is empty when the operator starts and isn't populated until the mons are done. Early on in working with Rook, I found it very confusing trying to understand which data fields of different structs were populated at any given time since some of them have fields that are populated at different times or left unpopulated in the case of some of the daemons.